rocAL/rocAL_pybind/amd/rocal/decoders.py

```diff
@@ -0,0 +1,264 @@
+
+    elif reader == "CaffeReader" or reader == "CaffeReaderDetection":
+        kwargs_pybind = {
+            "source_path": path,
+            "color_format": output_type,
+            "shard_id": shard_id,
+            "num_shards": num_shards,
+            'is_output': False,
+            "shuffle": random_shuffle,
+            "loop": False,
+            "decode_size_policy": types.MAX_SIZE,
+            "max_width": 0,
+            "max_height":0}
+        decoded_image = b.Caffe_ImageDecoderShard(Pipeline._current_pipeline._handle ,*(kwargs_pybind.values()))
+
+    else:
+        kwargs_pybind = {
+            "source_path": file_root,
+            "color_format": output_type,
+            "shard_id": shard_id,
+            "num_shards": num_shards,
+            'is_output': False,
+            "shuffle": random_shuffle,
+            "loop": False,
+            "decode_size_policy": types.USER_GIVEN_SIZE,
+            "max_width": 2000,
+            "max_height":2000}
+        decoded_image = b.ImageDecoderShard(Pipeline._current_pipeline._handle ,*(kwargs_pybind.values()))
+
+    return (decoded_image)
+
+
+def image_raw(*inputs, user_feature_key_map = None, path='', file_root ='', annotations_file= '', shard_id = 0, num_shards = 1, random_shuffle = False, affine=True, bytes_per_sample_hint=0, cache_batch_copy= True, cache_debug = False, cache_size = 0, cache_threshold = 0,
+                 cache_type='', device_memory_padding=16777216, host_memory_padding=8388608, hybrid_huffman_threshold= 1000000, output_type = types.RGB,
+                 preserve=False, seed=-1, split_stages=False, use_chunk_allocator= False, use_fast_idct = False, device = None):
+    reader = Pipeline._current_pipeline._reader
+
+    if (reader == "TFRecordReaderClassification" or reader == "TFRecordReaderDetection"):
+        kwargs_pybind = {
+            "source_path": path,
+            "user_key_for_encoded": user_feature_key_map["image/encoded"],
+            "user_key_for_filename": user_feature_key_map["image/filename"],
+            "color_format": output_type,
+            'is_output': False,
+            "shuffle": random_shuffle,
+            "loop": False,
+            "out_width": 2000,
+            "out_height": 2000}
+        decoded_image   = b.TF_ImageDecoderRaw(Pipeline._current_pipeline._handle ,*(kwargs_pybind.values()))
+        # decoded_image = b.TF_ImageDecoderRaw(handle, input_image, self._user_feature_key_map["image/encoded"], self._user_feature_key_map["image/filename"], , is_output, shuffle, False, decode_width, decode_height)
+        return (decoded_image)
+
+
+def image_random_crop(*inputs,user_feature_key_map=None ,path = '', file_root= '', annotations_file='', num_shards = 1, shard_id = 0, random_shuffle = False, affine=True, bytes_per_sample_hint=0, device_memory_padding= 16777216, host_memory_padding = 8388608, hybrid_huffman_threshold = 1000000,
```

Comment:
please remove unused arguments like device_memory_padding etc from this definition